frontend/src/views/Nodes/components/TNodesList.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div class="nodes-list">
+    <div class="nodes-list__heading">
+      <p class="nodes-list__heading-title">Name</p>
+      <p class="nodes-list__heading-title">IP</p>
+      <p class="nodes-list__heading-title">OS</p>
+      <p class="nodes-list__heading-title">Roles</p>
+      <p class="nodes-list__heading-title">Status</p>
+    </div>
+    <t-group-animation>
+      <t-nodes-item
+        v-for="item in filteredNodesList"
+        :key="item?.metadata?.name"
+        :item="item"
+      />
+    </t-group-animation>
+  </div>
+</template>
+
+<script>
+import TNodesItem from "./TNodesItem.vue";
+import TGroupAnimation from "@/components/common/Animation/TGroupAnimation.vue";
+import { computed, toRefs } from "@vue/reactivity";
+import { TNodesViewFilterOptions } from "@/constants";
+
+export default {
+  components: { TNodesItem, TGroupAnimation },
+  props: {
+    items: Object,
+    searchOption: String,
+    filterOption: String,
+  },
+  setup(props) {
+    const { items, searchOption, filterOption } = toRefs(props);
+
+    const getStatus = (item) => {
+      const conditions = item.status?.conditions;
+      if (!conditions) return "Not Ready";
+
+      for (const c of conditions) {
+        if (c["type"] === "Ready" && c["status"] === "True") return "Ready";
+      }
+
+      return "Not Ready";
```

Comment:
Would be nice to use `constants/index.ts` enum here.
Also you can extract this method as it's used both here and in `TNodesItem.vue`